operator/pkg/reconciliation/decommission_node.go

```diff
@@ -158,7 +157,12 @@ func (rc *ReconciliationContext) CheckDecommissioningNodes(epData httphelper.Cas
 
 	for _, pod := range rc.dcPods {
 		if pod.Labels[api.CassNodeState] == stateDecommissioning {
-			if !IsDoneDecommissioning(pod, epData) {
+			if !HasStartedDecommissioning(pod, epData) {
```

Comment:
Is there a risk here of having it get stuck whenever a pod finishes decommissioning? Like, if the pod status is `LEFT`, we will always enter this first if block right?

should it be changed to:
```
if !HasStartedDecommissioning... && !IsDoneDecommissioning...
```
or move the new block inside of the old one like:
``` 
if !IsDoneDecommissioning(pod, epData) {
   if !HasStartedDecommissioning....
```
instead?

Reply:
the name is maybe bad but no they are looking for mutually exclusive statuses 

func HasStartedDecommissioning(pod *v1.Pod, epData httphelper.CassMetadataEndpoints) bool {
	for idx := range epData.Entity {
		ep := &epData.Entity[idx]
		if ep.GetRpcAddress() == pod.Status.PodIP {
			return !strings.HasPrefix(ep.Status, "LEAVING")
		}
	}
	return true
}

func IsDoneDecommissioning(pod *v1.Pod, epData httphelper.CassMetadataEndpoints) bool {
	for idx := range epData.Entity {
		ep := &epData.Entity[idx]
		if ep.GetRpcAddress() == pod.Status.PodIP {
			return strings.HasPrefix(ep.Status, "LEFT")
		}
	}

Reply:
nice catch